src/tasks/tasks.js

```diff
@@ -53,23 +55,65 @@ function sleep(ms) {
   });
 }
 
-function dig() {
+function digTask(targetBlock) {
   return new Promise((resolve, reject) => {
-    if (bot.targetDigBlock) {
-      bot.chat('already digging ' + bot.targetDigBlock.name);
+      bot.dig(targetBlock, function () {
+        resolve();
+      });
+    });
+}
+
+function dig(position) {
+  position = position ? position : bot.entity.position.offset(0, -1, 0)
+  if (bot.targetDigBlock) {
+    bot.chat('already digging ' + bot.targetDigBlock.name);
+  } else {
+    var target = bot.blockAt(position);
+    if (target && bot.canDigBlock(target)) {
+      return digTask(target);
     } else {
-      var target = bot.blockAt(bot.entity.position.offset(0, -1, 0));
-      if (target && bot.canDigBlock(target)) {
-        bot.dig(target, function () {
-          resolve();
-        });
-      } else {
-        bot.chat('cannot dig');
-      }
+      bot.chat('cannot dig');
+      return Promise.resolve();
     }
-  }).catch(err => {
-    console.log(err.stack);
-  });
+  }
+}
+
+var digWallTask = function(height) {
+  if (height === 0) {
+    return Promise.resolve();
+  } else {
+    return dig(center(bot.entity.position.offset(1,height-1,0)))
+              .then(()=>{return digWallTask(height-1)});
+  }
+}
+
+function digStairTask(steps, height) {
+  // Height of 2-5
+  height = height === undefined? 2 : height;
+  height = Math.min(height, 5);
+  height = Math.max(height, 2);
+  console.log('height' + height);
+  if (steps === 0) {
+    console.log('done');
+    return Promise.resolve();
+  } else {
+    console.log('dig step ' + steps);
+
+    // Dig down
+    return dig().then(()=>{return sleep(1000)})
+
+                // // Dig front
+                // .then(()=>{return dig(center(bot.entity.position.offset(1,0,0)))})
+                // // Dig front and up
+                // .then(()=>{return dig(center(bot.entity.position.offset(1,1,0)))})
+                .then(()=>{return digWallTask(height)})
```

Comment:
When you have a single expression on the right side, you can leave out the `{}` and `return`.